python/cudf/dataframe/buffer.py

```diff
@@ -132,8 +132,8 @@ def __getitem__(self, arg):
             buf = Buffer(sliced)
             buf.dtype = self.dtype  # for np.datetime64 support
             return buf
-        elif isinstance(arg, int):
-            arg = utils.normalize_index(arg, self.size)
+        elif isinstance(arg, (int, np.integer)):
```

Comment:
This only catches integers defined as numpy integers and not just general Python integers. I don't imagine an end user will directly interact with the buffer objects, but I think we should still handle that case.

Reply:
I don't follow since `isinstance(arg, (int, np.integer))` will be `True` when `arg` is either a numpy integer or a Python `int`.
Can you be more specific on the phrase "This only catches..." ?